File: controllers/LabelController.ts
import { Label, NewLabel } from "@/models/Label";
import {
    createLabel,
    deleteLabel,
    subscribeToLabels,
} from "@/services/labelService";

export { subscribeToLabels };

export async function addLabel(
  userId: string,
  name: string,
  color: string,
): Promise<string> {
  const trimmed = name.trim();
  if (!trimmed) {
    throw new Error("Escribe un nombre para la etiqueta");
  }
  if (!color) {
    throw new Error("Selecciona un color");
  }
  const newLabel: NewLabel = {
    name: trimmed,
    color,
    userId,
  };
  return createLabel(userId, newLabel);
}

export async function removeLabel(
  userId: string,
  labelId: string,
): Promise<void> {
  await deleteLabel(userId, labelId);
}

export function getLabelById(
  labels: Label[],
  labelId: string,
): Label | undefined {
  return labels.find((l) => l.id === labelId);
}

export function getLabelsByIds(labels: Label[], labelIds: string[]): Label[] {
  return labelIds
    .map((id) => labels.find((l) => l.id === id))
    .filter(Boolean) as Label[];
}

File: src/main/kotlin/com/tasklife/controllers/LabelController.kt
package com.tasklife.controllers

import com.tasklife.models.Label
import com.tasklife.models.NewLabel
import com.tasklife.service.LabelService

class LabelController {

    private val service = LabelService()

    fun list(userId: String): List<Label> = service.list(userId)

    fun get(userId: String, labelId: String): Label? = service.get(userId, labelId)

    fun create(userId: String, label: NewLabel): Label = service.create(userId, label)

    fun delete(userId: String, labelId: String): Boolean = service.delete(userId, labelId)
}
